NitroxServer/GameLogic/PlayerManager.cs

```diff
@@ -252,13 +248,19 @@ public void PlayerDisconnected(NitroxConnection connection)
             }
         }
 
-        public void FinishProcessingReservation()
+        public void FinishProcessingReservation(Player player = null)
         {
             initialSyncTimer.Dispose();
             PlayerCurrentlyJoining = false;
 
             Log.Info($"Finished processing reservation. Remaining requests: {JoinQueue.Count}");
 
+            // If this is the first player connecting we need to restart time at this exact moment
+            if (player != null && ConnectedPlayers().Count() == 1)
```

Comment:
Why did we need to move this down to this method?  In what cases is the player object null?  Can this lead to us never resuming?

Reply:
First of all, we really don't want time to be passing when no player is actually fully loaded in the world because it would mess up the scheduled goals (and other story events based on time) if players were to execute it while loading.
Secondly, [``FinishProcessingReservation``](https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxServer/GameLogic/PlayerManager.cs#L255) can be called when a player finished connecting properly (from the ``PlayerSyncedFinished`` packet) or when a player failed to connect (from the same class, in [``InitialSyncTimerElapsed``](https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxServer/GameLogic/PlayerManager.cs#L155)) in which case the player object is null.
So we need time to resume at the exact moment that a player finished loading and this is the right place to detect it.
Also, ``FinishProcessingReservation`` will always be called because either the player successfully connects to the server and sends a packet that is received and calls the method, or if the packet is never received for any possible reason, the player joining timer will call this method.